flink-kubernetes/src/test/java/org/apache/flink/kubernetes/kubeclient/decorators/CmdJobManagerDecoratorTest.java

```diff
@@ -0,0 +1,97 @@
+import org.apache.flink.kubernetes.kubeclient.KubernetesJobManagerTestBase;
+import org.apache.flink.kubernetes.utils.Constants;
+import org.apache.flink.kubernetes.utils.KubernetesUtils;
+
+import org.junit.Test;
+
+import java.util.List;
+
+import static org.hamcrest.Matchers.containsInAnyOrder;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.not;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.fail;
+
+/** General tests for the {@link CmdJobManagerDecorator}. */
+public class CmdJobManagerDecoratorTest extends KubernetesJobManagerTestBase {
+
+    private CmdJobManagerDecorator cmdJobManagerDecorator;
+
+    @Override
+    protected void onSetup() throws Exception {
+        super.onSetup();
+
+        this.cmdJobManagerDecorator = new CmdJobManagerDecorator(kubernetesJobManagerParameters);
+    }
+
+    @Test
+    public void testContainerIsDecorated() {
+        flinkConfig.set(DeploymentOptions.TARGET, KubernetesDeploymentTarget.SESSION.getName());
+        final FlinkPod resultFlinkPod = cmdJobManagerDecorator.decorateFlinkPod(baseFlinkPod);
+        assertThat(
+                resultFlinkPod.getPodWithoutMainContainer(),
+                is(equalTo(baseFlinkPod.getPodWithoutMainContainer())));
+        assertThat(
+                resultFlinkPod.getMainContainer(), not(equalTo(baseFlinkPod.getMainContainer())));
+    }
+
+    @Test
+    public void testSessionClusterCommandsAndArgs() {
+        testJobManagerCommandsAndArgs(KubernetesDeploymentTarget.SESSION.getName());
+    }
+
+    @Test
+    public void testApplicationClusterCommandsAndArgs() {
+        testJobManagerCommandsAndArgs(KubernetesDeploymentTarget.APPLICATION.getName());
+    }
+
+    @Test
+    public void testUnsupportedDeploymentTargetShouldFail() {
+        try {
+            testJobManagerCommandsAndArgs("unsupported-deployment-target");
+            fail("Decorator should fail when unsupported deployment target is set.");
+        } catch (Exception ex) {
+            assertThat(ex, FlinkMatchers.containsCause(IllegalArgumentException.class));
+        }
+    }
+
+    private void testJobManagerCommandsAndArgs(String target) {
+        flinkConfig.set(DeploymentOptions.TARGET, target);
+        final FlinkPod resultFlinkPod = cmdJobManagerDecorator.decorateFlinkPod(baseFlinkPod);
+        final String entryCommand = flinkConfig.get(KubernetesConfigOptions.KUBERNETES_ENTRY_PATH);
+        assertThat(
+                resultFlinkPod.getMainContainer().getCommand(), containsInAnyOrder(entryCommand));
+        List<String> flinkCommands =
+                KubernetesUtils.getStartCommandWithBashWrapper(
+                        Constants.KUBERNETES_JOB_MANAGER_SCRIPT_PATH + " " + target);
+        assertThat(
+                resultFlinkPod.getMainContainer().getArgs(),
+                containsInAnyOrder(flinkCommands.toArray()));
```

Comment:
isn't the order somewhat important?

Reply:
Yes, the order should be checked. I will fix it.